Vivek/ElevationMap/ElevationMap.java

```diff
@@ -0,0 +1,75 @@
+package ElevationMap;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.IntStream;
+
+
+/**
+ * Imagine standing at an elevation,
+ * <p>
+ * The water that can be stored at that exact elevation would be between the highest elevation on the left and the highest elevation on the right.
+ * Needless to say, the height would be minimum of max left and max right.
+ * <p>
+ * Thus, total amount at any elevation = min(max left, max right) - current elevation.
+ * <p>
+ * Of course, total amount at any elevation would never be negative.
+ * <p>
+ * Hence, final formula for total amount at any elevation = max ( min(max left, max right) - current elevation, 0)
+ * <p>
+ * Total amount of water filled would be sum of total amount at all the elevations.
+ */
+public class ElevationMap {
+
+    public static void main(String[] args) {
```

Comment:
For some reason my brain has a really hard time visualizing this solution. Maybe you guys can help me understand in the meeting.